### PR TITLE
Jtso/ontology

### PR DESCRIPTION
**Pull request by jtsodapop**
files written under pep8 via yapf format updating

ontology_generator classes moved to ontology file

ontology tests updated and tested to all pass 

Screen Shot 2021-03-09 at 8 54 06 PM

### CODE REVIEW

File: labelbox/schema/ontology.py [hunk 1]
```diff
@@ -124,3 +216,45 @@ def convert_keys(json_dict: Dict[str, Any],
     if isinstance(json_dict, list):
         return [convert_keys(ele, converter) for ele in json_dict]
     return json_dict
+
+
+@dataclass
+class OntologyBuilder:
```
Comment (msokoloff1) [inline labelbox/schema/ontology.py:222]: Add a docstring to explain what this is and how it should be used. This is super confusing that an ontology is independent of a project yet you can only create one when you setup a project. This is a limitation of the backend but lets just give a little explanation for what this is and how it should be used.

**Review by msokoloff1 — CHANGES_REQUESTED** [review]
Looks pretty solid! I just have a few minor things that I would like to see changed. Also longer term we might want to support type checking. Right now there is still a chance of creating invalid ontologies even if you pass in the required fields.

File: labelbox/schema/ontology.py [hunk 2]
```diff
@@ -1,66 +1,158 @@
...
-        ]
-        return cls(**_dict)
+    def from_dict(cls, dictionary: Dict[str, Any]):
+        return Classification(
+            class_type=Classification.Type(dictionary["type"]),
+            instructions=dictionary["instructions"],
+            required=dictionary["required"],
+            options=[Option.from_dict(o) for o in dictionary["options"]],
+            schema_id=dictionary["schemaNodeId"],
+            feature_schema_id=dictionary["schemaNodeId"])
```
[the remark this answers is not in the document]
Comment (jtsodapop) [reply labelbox/schema/ontology.py:82]: change dictionary[] to dictionary.get(key) for optional

**Review by msokoloff1 — APPROVED** [review]
LGTM!

**Pull request merged**